packages/jest-matcher-utils/src/Replaceable.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import getType = require('jest-get-type');
+
+const supportTypes = ['map', 'array', 'object'];
+
+type ReplaceableForEachCallBack = (value: any, key: any, object: any) => void;
+
+export default class Replaceable {
+  object: any;
+  type: string;
+
+  constructor(object: any) {
+    this.object = object;
+    this.type = getType(object);
+    if (!supportTypes.includes(this.type)) {
+      throw new Error(`Type ${this.type} is not support in Replaceable!`);
+    }
+  }
+
+  static isReplaceable(obj1: any, obj2: any): boolean {
+    const obj1Type = getType(obj1);
+    const obj2Type = getType(obj2);
+    return obj1Type === obj2Type && supportTypes.includes(obj1Type);
+  }
+
+  forEach(cb: ReplaceableForEachCallBack): void {
+    if (this.type === 'object') {
+      Object.entries(this.object).forEach(([key, value]) => {
+        cb(value, key, this.object);
+      });
```

Comment:
> The `Object.entries()` method returns an array of a given object's own enumerable **string-keyed** property `[key, value]` pairs

See [`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) on Mozilla Developer Network

Therefore, you might need to add the following iteration for enumerable **symbol-keyed** properties:

```js
  Object.getOwnPropertySymbols(this.object).forEach(key => {
    const descriptor = Object.getOwnPropertyDescriptor(this.object, key);
    if ((descriptor as PropertyDescriptor).enumerable) {
      cb(this.object[key], key, this.object);
    }
  });
```

The `forEach` method for `Map` and `Set` do not have this problem.

Reply:
None of the diff snapshots in `matchers.test.ts.snap` cover the case of symbol keys :(

Reply:
I will add some symbol keys test cases, and make them work.
Thanks.

Reply:
@WeiAnAn For your information, the current target for Jest 25 is Tuesday January 21, so if this rare edge case will slow you down too much, please leave it for a follow-up pull request. The copy function is the important thing.